core/src/io/anuke/mindustry/world/blocks/distribution/ArmoredConduit.java

```diff
@@ -0,0 +1,45 @@
+package io.anuke.mindustry.world.blocks.distribution;
+
+import io.anuke.arc.Core;
+import io.anuke.arc.graphics.g2d.Draw;
+import io.anuke.arc.graphics.g2d.TextureRegion;
+import io.anuke.mindustry.type.Liquid;
+import io.anuke.mindustry.world.Block;
+import io.anuke.mindustry.world.Edges;
+import io.anuke.mindustry.world.Tile;
+
+public class ArmoredConduit extends Conduit{
+    protected TextureRegion capRegion;
+
+    public ArmoredConduit(String name){
+        super(name);
+        leakResistance = 10f;
+    }
+
+    @Override
+    public void load(){
+        super.load();
+        capRegion = Core.atlas.find(name + "-cap");
+    }
+
+    @Override
+    public void draw(Tile tile){
+        super.draw(tile);
+
+        // draw the cap when a conduit would normally leak
+        Tile next = tile.getNearby(tile.rotation());
```

Comment:
Use your `front()` method so links are accounted for properly. Also make sure it's not null.